src/huggingface_hub/inference/_client.py

```diff
@@ -240,6 +240,10 @@ def post(
                 hf_raise_for_status(response)
                 return response.iter_lines() if stream else response.content
             except HTTPError as error:
+                if error.response.status_code == 422 and task is not None:
+                    error.args = (
+                        error.args[0] + f"\nMake sure '{task}' task is supported by the model.",
```

Comment:
```suggestion
                        f"{error.args[0]}\nMake sure '{task}' task is supported by the model.",
```

I think f-str is a bit easier than concatenation. Also, why does sync client delimit with `"\n"` and async client delimit with `". "`?

Reply:
Thanks for the review! f-string makes sense yes :+1: 

> Also, why does sync client delimit with "\n" and async client delimit with ". "?

For stylish reasons. The sync client raises a `requests.HTTPError` exception which print the message when serialized. I prefer the `\n` so that the warning is on its own line. For the async client we have a `aiohttp.ClientResponseError` which doesn't serialize the same way (see snippet). So I preferred to simple put `". "`.
But no strong opinion on it ^^